### PR TITLE
feat: login form

### LINKED ISSUE #5: create login page form
Create landing page form that will match these designs: ***[Figma - landing page](https://www.figma.com/file/IVY7iX8XnrzpiUNH8aJOz6/dorm-social-rooms-manager?node-id=5504%3A69519&t=hOCtTKVDCczmQqkp-0)***

Form should be used in Login page

Form should be responsive and handle 3 breakpoints:

 - **360px** >= screen width < **768px** → mobile
 - **768px** >= screen width <  **1024px** → tablet
 - **1200px** >= screen width → desktop
 
### Requirements

- All fields are required to fill
- Form checkbox **Remember me** should do nothing
- Form link **Forgot password** should do nothing
- In password field after user enters any text, an eye icon should appear. 
  - If user click eye icon, password should become visible
  - If user click eye icon next time, text should be replaced with dots/start again
- If user filled whole form and clicked enter button, Login action should be triggered
- Login button should be disabled if form is not fully filled
  - If user filled form, login button should become enabled
- Register button should be always enabled and after click should redirect to Register page
- All text displayed on the webpage should be taken from `i18n`

### Resources:

- ***[Link](https://mui.com/material-ui/react-link/)***
- ***[Text Field](https://mui.com/material-ui/react-text-field/)***
- ***[Checkbox](https://mui.com/material-ui/react-checkbox/)***
- ***[Icon](https://mui.com/material-ui/icons/)***
- ***[Button](https://mui.com/material-ui/react-button/#main-content)***
- ***[formik](https://formik.org/docs/examples/with-material-ui)***

### PR DESCRIPTION
**Pull request by 235923**
Closes #5

### CODE REVIEW

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
- [ ] You need to use formik to handle forms instead of multiple useState
- [ ] Creating such components as `LoginButton` or `RegisterButton` doesn't make any sense because you will be able to use them only once in this specific form. Please do not create such small components that you will use in only one place it makes code much more complexed and harder to read

**FYI:** Previously, we have also discussed that you should create a single component that will take an object and auto-generate form from given fields in the object, but let's skip this for now. You can keep Login and Register forms as separated components for now.

File: src/components/LoginForm/LoginForm.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+interface buttonFunctions {
+  RegisterButtonFunction: MouseEventHandler<HTMLButtonElement>;
+  LoginButtonFunction: MouseEventHandler<HTMLButtonElement>;
+}
+
+export function LoginForm({ LoginButtonFunction, RegisterButtonFunction }: buttonFunctions) {
+  const theme = useTheme();
+  const isMobile: boolean = useMediaQuery(theme.breakpoints.down('tablet'));
+  const [pass, setPass] = useState('');
+  const [email, setEmail] = useState('');
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: useStates replaced by formik in 9b6ca91e67ac23a241b7f845f5160ef1a6045c12

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: src/components/LoginForm/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import { LoginOptions } from '../LoginOptions/LoginOptions';
+import { MouseEventHandler } from 'react';
+
+interface buttonFunctions {
+  RegisterButtonFunction: MouseEventHandler<HTMLButtonElement>;
+}
+type Err = {
+  email?: string;
+  password?: string;
+};
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: added separate file in d21c8f218ec39c416608d58f27fc54b5c27fe7bc

File: src/components/LoginForm/LoginForm.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+export function LoginForm({ RegisterButtonFunction }: buttonFunctions) {
+  const theme = useTheme();
+  const isMobile: boolean = useMediaQuery(theme.breakpoints.down('tablet'));
+  const minLenght = 1;
+
+  const validate = (values: { email: string; password: string }) => {
+    const errors: Err = {};
+    if (values.email.length < minLenght) {
+      errors.email = 'Email is required';
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: added i18n in 53b3e914920b91b4b278d8b05529171eddad3e2b

File: src/components/LoginForm/LoginForm.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    }
+    if (values.password.length < minLenght) {
+      errors.password = 'Password is required';
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: added i18n in 53b3e914920b91b4b278d8b05529171eddad3e2b

File: src/components/PasswordInput/PasswordInput.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  const [showPassword, setshowPassword] = useState(false);
+  function handleShowPassword() {
+    setshowPassword(!showPassword);
+  }
+
+  return (
+    <Field
+      as={TextField}
+      className='input'
+      label='Password'
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/PasswordInput/PasswordInput.tsx]: added i18n in 53b3e914920b91b4b278d8b05529171eddad3e2b

File: src/components/PasswordInput/PasswordInput.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      type={showPassword ? 'text' : 'password'}
+      name='password'
+      required
+      InputProps={{
+        endAdornment: (
+          <InputAdornment position='end'>
+            <IconButton
+              aria-label='toggle password visibility'
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/PasswordInput/PasswordInput.tsx]: added i18n in 53b3e914920b91b4b278d8b05529171eddad3e2b

File: src/components/LoginInputs/LoginInputs.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      spacing={3}
+    >
+      <Grid
+        item
+        base={12}
+      >
+        <Field
+          as={TextField}
+          className='input'
+          label='e-mail'
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginInputs/LoginInputs.tsx]: added i18n in 53b3e914920b91b4b278d8b05529171eddad3e2b

File: src/components/LoginOptions/LoginOptions.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+          control={<Checkbox defaultChecked />}
+          label='Remeber me'
+        />
+      </Grid>
+      <Grid
+        item
+        mobile={5}
+        base={8}
+      >
+        <Link href='#'>Forgot Password</Link>
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginOptions/LoginOptions.tsx]: added i18n in 53b3e914920b91b4b278d8b05529171eddad3e2b

File: src/components/LoginOptions/LoginOptions.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+      spacing={1}
+    >
+      <Grid
+        item
+        mobile={7}
+        base={8}
+      >
+        <FormControlLabel
+          control={<Checkbox defaultChecked />}
+          label='Remeber me'
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginOptions/LoginOptions.tsx]: added i18n in 53b3e914920b91b4b278d8b05529171eddad3e2b

File: src/components/LoginForm/LoginForm.tsx [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+                  )}
+                  <Grid
+                    item
+                    base={8}
+                  >
+                    <Button
+                      sx={{
+                        ':hover': {
+                          bgcolor: theme.palette.primary.main,
+                          color: 'red',
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: used theme in d21c8f218ec39c416608d58f27fc54b5c27fe7bc

File: src/components/LoginForm/LoginForm.tsx [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+                        },
+                        bgcolor: theme.palette.primary.main,
+                        color: 'white',
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: used theme in d21c8f218ec39c416608d58f27fc54b5c27fe7bc

File: src/components/LoginForm/LoginForm.tsx [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+                  </Grid>
+                  <Grid
+                    item
+                    base={8}
+                  >
+                    <Button
+                      sx={{
+                        ':hover': {
+                          bgcolor: theme.palette.secondary.main,
+                          color: 'red',
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: used theme in d21c8f218ec39c416608d58f27fc54b5c27fe7bc

File: src/components/LoginForm/LoginForm.tsx [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+                        },
+                        bgcolor: theme.palette.secondary.main,
+                        color: 'white',
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: used theme in d21c8f218ec39c416608d58f27fc54b5c27fe7bc

File: src/components/LoginForm/LoginForm.tsx [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+                        padding: 1,
+                        width: 1,
+                      }}
+                      onClick={RegisterButtonFunction}
+                      disabled={!true}
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: deleted in d21c8f218ec39c416608d58f27fc54b5c27fe7bc

File: src/components/LoginForm/LoginForm.tsx [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+  RegisterButtonFunction: MouseEventHandler<HTMLButtonElement>;
+}
+type Err = {
+  email?: string;
+  password?: string;
+};
+export function LoginForm({ RegisterButtonFunction }: buttonFunctions) {
+  const theme = useTheme();
+  const isMobile: boolean = useMediaQuery(theme.breakpoints.down('tablet'));
+  const minLenght = 1;
```
[the remark this answers is not in the document]
Comment (235923) [reply src/components/LoginForm/LoginForm.tsx]: fixed in e4a2ffc7ee8f95d99422a0220dbc51ba9aad2b2b

File: src/components/LoginForm/LoginForm.tsx [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+                    <Button
+                      variant='contained'
+                      color='secondary'
+                      sx={{
+                        padding: 1,
+                        width: 1,
+                      }}
+                      onClick={RegisterButtonFunction}
+                    >
+                      {t('LoginForm.register')}
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginForm.tsx:116]: you should use this:
https://mui.com/material-ui/guides/routing/#link

File: src/i18n/i18n.ts [hunk 17]
```diff
@@ -0,0 +1,28 @@
+import * as LoginFormEN from '../locales/en/LoginForm.json';
+import * as LoginFormPL from '../locales/pl/LoginForm.json';
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+
+const resources = {
+  en: {
+    components: {
+      LoginForm: LoginFormEN,
```
Comment (kaczor6418) [inline src/i18n/i18n.ts:9]: please use camelCase -> loginForm

File: src/i18n/i18n.ts [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+    },
+  },
+  pl: {
+    components: {
+      LoginForm: LoginFormPL,
```
Comment (kaczor6418) [inline src/i18n/i18n.ts:14]: please use camelCase -> loginForm

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
almost there 😄

File: src/components/Login/Login.tsx [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+export function Login() {
+  const [isRegister, setIsRegister] = useState(false);
+  function RegisterClick() {
+    setIsRegister(!isRegister);
+  }
+  return (
+    <ThemeProvider theme={theme}>
+      <div>{!isRegister && <LoginForm RegisterButtonFunction={RegisterClick} />}</div>
```
Comment (kaczor6418) [inline src/components/Login/Login.tsx:37]: I do not understand this, why do we need this `!isRegister &&`? + I think Login should be inside /pages/LoginPage.tsx

File: src/components/LoginInputs/LoginInputs.tsx [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+      </Grid>
+      <Grid
+        item
+        base={12}
+      >
+        <PasswordInput />
+      </Grid>
+    </Grid>
+  );
+}
```
Comment (kaczor6418) [inline src/components/LoginInputs/LoginInputs.tsx:38]: you will never reuse this component, let's move this to `LoginForm`

File: src/components/LoginOptions/LoginOptions.tsx [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+      </Grid>
+      <Grid
+        item
+        mobile={5}
+        base={8}
+      >
+        <Link href='#'>{t('LoginForm.forgot_password')}</Link>
+      </Grid>
+    </Grid>
+  );
```
Comment (kaczor6418) [inline src/components/LoginOptions/LoginOptions.tsx:37]: you will never reuse this component, let's move this to `LoginForm`

File: src/components/PasswordInput/PasswordInput.tsx [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+export function PasswordInput() {
+  const [showPassword, setshowPassword] = useState(false);
+  function handleShowPassword() {
+    setshowPassword(!showPassword);
+  }
+
+  return (
+    <Field
+      as={TextField}
+      className='input'
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply src/components/PasswordInput/PasswordInput.tsx:18]: still not fixed

File: src/components/PasswordInput/PasswordInput.tsx [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { Field } from 'formik';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+export function PasswordInput() {
+  const [showPassword, setshowPassword] = useState(false);
+  function handleShowPassword() {
+    setshowPassword(!showPassword);
+  }
```
Comment (kaczor6418) [inline src/components/PasswordInput/PasswordInput.tsx:12]: let use arrow functions for callbacks

```javascript
const toggleShowPassword = () => setShowPassword((currentState) => !currentState)
```

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: src/components/LoginOptions/LoginOptions.tsx (hunk 21 above)
Comment (235923) [reply src/components/LoginOptions/LoginOptions.tsx]: it's used two times in LoginForm

File: src/components/LoginForm/LoginForm.tsx [hunk 24]
```diff
@@ -0,0 +1,162 @@
...
+                height={540}
+                alignItems='center'
+                justifyContent='center'
+              >
+                <Grid
+                  container
+                  paddingTop={{
+                    desktop: 8,
+                    mobile: 6,
+                    tablet: 8,
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginForm.tsx:60]: This is not needed.
you can simplify check to change layout if we reach table+ resolution:
```tsx
const isTablet = useMediaQuery(theme.breakpoints.up('tablet'));

return (
    <Grid
      container
      paddingTop={isTablet ? 8 : 6}

```

**FYI**: if you put this styles into *`.scss`* files then you do not need to worry about eslint-magic-number error 😉

File: src/App.tsx [hunk 25]
```diff
@@ -1,13 +1,13 @@
...
+import { LoginPage } from './pages/LoginPage';
 
 export function App() {
   return (
     <Routes>
       <Route
-        path='/'
-        element={<Footer />}
-      ></Route>
+        path='/login'
```
Comment (kaczor6418) [inline src/App.tsx:8]: path for `LoginPage` should be `/`

File: src/components/LoginForm/LoginForm.tsx [hunk 26]
```diff
@@ -0,0 +1,162 @@
+import '../../i18n/i18n.ts';
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginForm.tsx:1]: this should be imported once only in the root of our application

File: src/components/LoginForm/LoginForm.types.ts [hunk 27]
```diff
@@ -0,0 +1,4 @@
+export type LoginFormErrors = {
+  Email?: string;
+  Password?: string;
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginForm.types.ts:3]: please use `camelCase` notation for field names of interfaces/types

File: src/components/LoginForm/LoginOptions.tsx [hunk 28]
```diff
@@ -0,0 +1,38 @@
+import '../../i18n/i18n.ts';
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginOptions.tsx:1]: this should be imported once only in the root of our applciation

File: src/components/PasswordInput/PasswordInput.tsx [hunk 29]
```diff
@@ -0,0 +1,41 @@
+import '../../i18n/i18n.ts';
```
Comment (kaczor6418) [inline src/components/PasswordInput/PasswordInput.tsx:1]: this should be imported once only in the root of our applciation

File: src/components/LoginForm/LoginOptions.tsx [hunk 30]
```diff
@@ -0,0 +1,38 @@
...
+      >
+        <FormControlLabel
+          control={<Checkbox defaultChecked />}
+          label={t('loginForm.rememberMe')}
+        />
+      </Grid>
+      <Grid
+        item
+        mobile={5}
+        base={8}
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginOptions.tsx:32]: this is not needed

File: src/components/LoginForm/LoginOptions.tsx [hunk 31]
```diff
@@ -0,0 +1,38 @@
...
+        tablet: 'row',
+      }}
+      alignItems='center'
+      justifyContent='center'
+      spacing={1}
+    >
+      <Grid
+        item
+        mobile={7}
+        base={8}
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginOptions.tsx:22]: this is not needed

File: src/components/LoginForm/LoginOptions.tsx [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+export function LoginOptions() {
+  const { t } = useTranslation();
+  return (
+    <Grid
+      container
+      direction={{
+        desktop: 'row',
+        mobile: 'column',
+        tablet: 'row',
+      }}
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginOptions.tsx:14]: This is not needed.
you can simplify check to change layout if we reach table+ resolution:
```tsx
const isTablet = useMediaQuery(theme.breakpoints.up('tablet'));

return (
    <Grid
      container
      direction={isTablet ? 'row' : 'column'}

```

File: src/components/LoginForm/LoginForm.tsx [hunk 33]
```diff
@@ -0,0 +1,162 @@
...
+                padding={0}
+                boxShadow={'0px 0px 30px #ccc'}
+                maxWidth={{
+                  desktop: 400,
+                  mobile: 330,
+                  tablet: 400,
+                }}
+                height={540}
+                alignItems='center'
+                justifyContent='center'
```
Comment (kaczor6418) [inline src/components/LoginForm/LoginForm.tsx:53]: This is not needed

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
Almost :sunglasses:

**Review by kaczor6418 — APPROVED** [review]
LGTM 😎

**Pull request merged**